pybamm/models/submodels/pe_degradation/base_phase_transition.py

```diff
@@ -0,0 +1,265 @@
+#
+# Base class for the core-shell model for Positive Electrode (NMC811)
+# phase transition cased degradation, based on paper
+#    Mingzhao Zhuo, Gregory Offer, Monica Marinescu, "Degradation model of
+#    high-nickel positive electrodes: Effects of loss of active material and
+#    cyclable lithium on capacity fade", Journal of Power Sources,
+#    556 (2023): 232461. doi: 10.1016/j.jpowsour.2022.232461.
+#
+import pybamm
+
+
+class BasePhaseTransition(pybamm.BaseSubModel):
+    """
+    Base class for the core-shell model.
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    domain : str, optional
+        The domain of the model (default is "Positive")
+    options: dict
+        A dictionary of options to be passed to the model.
+        See :class:`pybamm.BaseBatteryModel`
+    phase : str, optional
+        Phase of the particle (default is "primary")
+
+    References
+    ----------
+    .. [1] Mingzhao Zhuo, Gregory Offer, Monica Marinescu, "Degradation model of
+           high-nickel positive electrodes: Effects of loss of active material
+           and cyclable lithium on capacity fade", Journal of Power Sources},
+           556 (2023): 232461.
+           doi: 10.1016/j.jpowsour.2022.232461.
+    """
+
+    def __init__(self, param, domain, options, phase="primary"):
+        super().__init__(param, domain, options=options, phase=phase)
+
+        # Read from options to see if we have a particle size distribution
+        domain_options = getattr(self.options, domain)
+        self.size_distribution = domain_options["particle size"] == "distribution"
+        if self.size_distribution is True:
```

Comment:
can this be caught earlier as an option check?

Reply:
Agreed